### PR TITLE
Object-oriented interface

### PR DESCRIPTION
**Pull request by jemiahlee**
Hey @bhardin,

Pursuant to our conversation from a couple of weeks ago, how's about you check out the path I've taken here with the gem? I haven't made classes for absolutely everything, and there are undoubtedly some mop-up tasks to be done here, but I wanted to get this in front of you. I also haven't modified the documentation, so you will probably have to read the tests to get a sense of what's really going on here. I will happily do so before we merge this in.

There are a lot of data file changes, sorry. That's a lot of the noise in this PR. lib/bgg-api is back to being small and neat though! :smiley:

### CODE REVIEW

File: lib/bgg_collection.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class BggCollection
```
Comment (bhardin) [inline lib/bgg_collection.rb]: I think I would modulize this.

```
module BGG
  class Collection
```

That way everything doesn't start with Bgg... I would also put this in `lib/bgg/collection.rb`

File: lib/bgg_collection_item.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
+class BggCollectionItem
```
Comment (bhardin) [inline lib/bgg_collection_item.rb]: Should this be a `BGG::Item` or a `BGG::Collection::Item` ?

File: lib/bgg_game.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+class BggGame
```
Comment (bhardin) [inline lib/bgg_game.rb]: Same... `BGG::Game`

**Comment by bhardin** [comment]
Let us make this Modular!

I can help later today if you would like. I like where this is heading though.

File: lib/bgg_collection_item.rb (hunk 2 above)
Comment (jemiahlee) [inline lib/bgg_collection_item.rb]: probably a private class inside BGG::Collection. Users shouldn't be instantiating these. The same way the search results are done (and the search iterator should be private too).

**Comment by bhardin** [comment]
Would you like me to try my hand at modularizing these? or do you want to do it?

**Comment by jemiahlee** [comment]
I'll do it. I just didn't have time last night.

**Comment by bhardin** [comment]
kk.

**Pull request closed without merging**